Modules/_sqlite/connection.c

```diff
@@ -33,6 +33,32 @@
 #define HAVE_TRACE_V2
 #endif
 
+static int
+clinic_fsconverter(PyObject *pathlike, const char **result)
+{
+    PyObject *bytes = NULL;
+    Py_ssize_t len;
+    char *str;
+
+    if (!PyUnicode_FSConverter(pathlike, &bytes)) {
+        goto error;
+    }
+    else if (PyBytes_AsStringAndSize(bytes, &str, &len) < 0) {
+        goto error;
+    }
+    else if ((*result = (const char *)PyMem_Malloc(len+1)) == NULL) {
+        goto error;
+    }
```

Comment:
For simplicity, if-else if + goto combination is too complex

```suggestion
    if (!PyUnicode_FSConverter(pathlike, &bytes)) {
        goto error;
    }
    if (PyBytes_AsStringAndSize(bytes, &str, &len) < 0) {
        goto error;
    }
    if ((*result = (const char *)PyMem_Malloc(len+1)) == NULL) {
        goto error;
    }
```

Reply:
> For simplicity, if-else if + goto combination is too complex

Doh, of course! It helps with a second pair of eyes :)